app/portainer/services/api/stackService.js

```diff
@@ -245,28 +256,39 @@ function StackServiceFactory($q, Stack, ResourceControlService, FileUploadServic
     return deferred.promise;
   };
 
-  service.createComposeStackFromFileContent = function(name, stackFileContent, env, endpointId) {
+  service.createStackFromFileContent = function(stackDeployOptions, stackFileContent) {
```

Comment:
I don't like the introduction of this function to determine which type of stack should be created, IMO we should keep the previous behaviour i.e. let the controller determine which service function to call.